### PR TITLE
Add setup command

### PR DESCRIPTION
**Pull request by masa213f**
https://github.com/cybozu-go/github-actions-controller/issues/19

Enable to specify a setup command and a slack channel in RunnerPool CR.

Signed-off-by: Masayuki Ishii <masa213f@gmail.com>

### CODE REVIEW

File: docs/crd-runner-pool.md [hunk 1]
```diff
@@ -15,12 +15,21 @@ GitHub Actions self-hosted runners.
...
+| `slackAgent`     | [SlackAgentConfig](#SlackAgentConfig)           | Configuration of a Slack agent.                          |
+| `template`       | [RunnerPodTemplateSpec](#RunnerPodTemplateSpec) | Pod manifest Template.                                   |
+
+SlackAgentConfig
+----------------
+
+| Field         | Type   | Description                                                                                                                                    |
+| ------------- | ------ | ---------------------------------------------------------------------------------------------------------------------------------------------- |
+| `serviceName` | string | Service name of Slack agent.                                                                                                                   |
+| `channel`     | string | Slack channel which the job results are reported. If this field is omitted, the default channel specified in slack-agent options will be used. |
```
Comment (zoetrope) [inline docs/crd-runner-pool.md:32]: It is difficult for me to find what `slack-agent options` is.
Please make it clear that it is the `-c` option of slack-agent command.

File: kindtest/workflows/job-failure.tmpl.yaml [hunk 2]
```diff
@@ -7,6 +7,6 @@ jobs:
     runs-on: [self-hosted, "{{.Namespace}}/{{.RunnerPool}}"]
     steps:
       - run: job-started
-      - run: false
-      - if: failure()
-        run: job-failure
+      # /tmp/test is created by a setup command in runnerpool2.
+      - run: cat /tmp/test
```
Comment (kmdkuk) [inline kindtest/workflows/job-failure.tmpl.yaml:11]: I am uncomfortable with checking if job-failure is being executed in a workflow that is expected to succeed.

File: docs/crd-runner-pool.md (hunk 1 above)
Comment (masa213f) [reply docs/crd-runner-pool.md]: You're right. Thank you.

File: kindtest/workflows/job-failure.tmpl.yaml (hunk 2 above)
Comment (masa213f) [reply kindtest/workflows/job-failure.tmpl.yaml]: Thank you for your comment.
In my opinion, the purpose of the `job-success/failure` workflow is to check the runner pod's behavior when `job-success/failure` is called.
So we need not fail a job on purpose. 

By the way, it was not good to have two checks in the `job-failure` workflow.
So I make another workflow `setup-command`.

Comment (masa213f) [reply kindtest/workflows/job-failure.tmpl.yaml]: Now the kindest implementation is redundant.
So I want to refactor it in another PR.

**Review by kmdkuk — APPROVED** [review]
LGTM

**Review by zoetrope — APPROVED** [review]
LGTM

**Pull request merged**